protocols/noise/src/io.rs

```diff
@@ -26,33 +26,17 @@ use futures::ready;
 use futures::prelude::*;
 use log::{debug, trace};
 use snow;
-use std::{fmt, io, pin::Pin, ops::DerefMut, task::{Context, Poll}};
+use std::{cmp::min, fmt, io, pin::Pin, ops::DerefMut, task::{Context, Poll}};
 
+/// Max. size of a noise package.
 const MAX_NOISE_PKG_LEN: usize = 65535;
-const MAX_WRITE_BUF_LEN: usize = 16384;
-const TOTAL_BUFFER_LEN: usize = 2 * MAX_NOISE_PKG_LEN + 3 * MAX_WRITE_BUF_LEN;
+/// Extra space given to the encryption buffer to hold key material.
+const EXTRA_ENCRYPT_SPACE: usize = 1024;
```

Comment:
That constant looks a bit like dark magic to me, but the test would cover any possible mistake here anyway.